BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/PortalsManager.java

```diff
@@ -105,33 +107,38 @@ private void removeTeleportedPlayers(){
 
 					@Override
 					public void run() {
-						List<Player> toRemove = new ArrayList<Player>();
-						for (Player p: arrivedPlayers){
-							if (!p.isOnline()) {
-								toRemove.add(p);
+						List<UUID> toRemove = new LinkedList<UUID>();
+						long currTime = System.currentTimeMillis();
+						for (Entry <UUID, Long> entry: arrivedPlayers.entrySet()){
+							Player p = Bukkit.getPlayer(entry.getKey());
+							if (p == null) {
 								continue;
```

Comment:
So that kind of caching would persist until restart; useful but good to note its limits.

So teleport actually lands you inside the other side's portal? Not outside of it?